core/src/autogluon/core/models/ensemble/bagged_ensemble_model.py

```diff
@@ -199,7 +200,11 @@ def _fit(self, X_train, y_train, k_fold=5, k_fold_start=0, k_fold_end=None, n_re
                 fold_model = copy.deepcopy(model_base)
                 fold_model.name = f'{fold_model.name}S{j+1}F{fold_num_in_repeat+1}'  # S5F3 = 3rd fold of the 5th repeat set
                 fold_model.set_contexts(self.path + fold_model.name + os.path.sep)
-                fold_model.fit(X_train=X_train_fold, y_train=y_train_fold, X_val=X_val_fold, y_val=y_val_fold, time_limit=time_limit_fold, **kwargs)
+                kwargs_fold = kwargs.copy()
+                if sample_weights is not None:
+                    kwargs_fold['sample_weights'] = sample_weights[train_index]
```

Comment:
bug:

`sample_weights.iloc`, otherwise could be misaligned:

```
X_train_fold, X_val_fold = X_train.iloc[train_index, :], X_train.iloc[val_index, :]
y_train_fold, y_val_fold = y_train.iloc[train_index], y_train.iloc[val_index]
```

Reply:
Actually sample_weight here is numpy array, so the integer indexing should already be correct